### PR TITLE
Fix Callbacks

### PR DESCRIPTION
**Pull request by kanav99**

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,32 +1,34 @@
...
-Compat = "34da2185-b29b-5c13-b0c7-acf172513d20"
-DiffEqOperators = "9fdde737-9c7f-55bf-ade8-46b3f136cc48"
-Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
-
-[targets]
-test = ["Test", "DiffEqProblemLibrary", "ForwardDiff", "Compat", "DiffEqOperators"]
+name = "Sundials"
+uuid = "c3572dad-4567-51f8-b174-8c6c989267f4"
+authors = ["Chris Rackauckas <accounts@chrisrackauckas.com>"]
+version = "3.8.1"
```
Comment (YingboMa) [inline Project.toml]: Why is the version number dropped?

Comment (kanav99) [reply Project.toml]: Oops sorry about that. I just `dev`ed Sundials just now and still got such issue

File: Project.toml [hunk 2]
```diff
@@ -1,32 +1,34 @@
...
-Compat = "34da2185-b29b-5c13-b0c7-acf172513d20"
-DiffEqOperators = "9fdde737-9c7f-55bf-ade8-46b3f136cc48"
-Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
-
-[targets]
-test = ["Test", "DiffEqProblemLibrary", "ForwardDiff", "Compat", "DiffEqOperators"]
+name = "Sundials"
+uuid = "c3572dad-4567-51f8-b174-8c6c989267f4"
+authors = ["Chris Rackauckas <accounts@chrisrackauckas.com>"]
+version = "3.8.2"
```
Comment (YingboMa) [inline Project.toml]:
```suggestion
version = "3.8.3"
```
We need a new patch release.

File: src/common_interface/solve.jl [hunk 3]
```diff
@@ -62,6 +62,8 @@ function DiffEqBase.__init(
 
     progress && @logmsg(-1,progress_name,_id=_id = :Sundials,progress=0)
 
+    uBottomEltype = recursive_bottom_eltype(prob.u0)
```
Comment (ChrisRackauckas) [inline src/common_interface/solve.jl]: even better: we know it has to be Float64

Comment (kanav99) [reply src/common_interface/solve.jl]: Lol let me redo this whole PR

**Comment by ChrisRackauckas** [comment]
Can we get a test?

**Comment by kanav99** [comment]
There are some issues with Sundials + VectorContinuousCallbacks. I will check this today

**Comment by kanav99** [comment]
Nope there wasn't any issue. It was just me being an idiot. The PR is good to go!

**Pull request merged**